### PR TITLE
close #687 empty prefix is allowed

### LINKED ISSUE #687: The class creation adds a backslash on empty spec prefix
When the prefix is set as empty the spec class creation uses a backslash prefixed namespace.

Our use case is the following, we do not want to split the namespaces for the specs and the source code, this way we simplify a lot of imports, we have separated the specs and source code by folders, we do not need to separate by namespaces, the specs are already suffixed on the classes, so we do not need to prefix by namespace too.

### PR DESCRIPTION
**Pull request by Markcial**
With this PR the spec prefix can be empty, this way the namespace does not start with a backslash, which is a not valid class name.

### CODE REVIEW

File: spec/PhpSpec/Locator/PSR0/PSR0LocatorSpec.php [hunk 1]
```diff
@@ -53,6 +53,16 @@ function it_generates_proper_fullSrcPath_even_from_empty_namespace()
         $this->getFullSrcPath()->shouldReturn(dirname(__DIR__).DIRECTORY_SEPARATOR);
     }
 
+    function it_should_not_have_backslash_on_missing_prefix()
+    {
+        $this->beConstructedWith('Cust\Ns', '', dirname(__DIR__), __DIR__);
+
+        $this->getSpecNamespace()->shouldReturn('Cust\Ns\\');
+        $this->getSpecNamespace()->shouldNotReturn('\Cust\Ns\\');
```
Comment (ciaranmcnulty) [inline spec/PhpSpec/Locator/PSR0/PSR0LocatorSpec.php]: If it returns the other value you don't need to specify a value it doesn't return

File: spec/PhpSpec/Locator/PSR0/PSR0LocatorSpec.php [hunk 2]
```diff
@@ -53,6 +53,16 @@ function it_generates_proper_fullSrcPath_even_from_empty_namespace()
...
+
+        $this->getFullSpecPath()->shouldReturn(__DIR__.DIRECTORY_SEPARATOR.'Cust'.DIRECTORY_SEPARATOR.'Ns'.DIRECTORY_SEPARATOR);
```
Comment (ciaranmcnulty) [inline spec/PhpSpec/Locator/PSR0/PSR0LocatorSpec.php]: This should perhaps be a different example

**Comment by ciaranmcnulty** [comment]
This maybe could do with a `.feature` illustrating the impact this has on user behaviour

**Comment by Markcial** [comment]
Just a moment please, i will update the PR with a feature instead of a spec test

**Comment by ciaranmcnulty** [comment]
No keep the spec too ;-)

**Comment by ciaranmcnulty** [comment]
Thanks for the feature, it is a lot clearer to me now what this PR is about :+1:

**Comment by Markcial** [comment]
Oh sorry, i will add again the spec then. Thanks to you (the phpspec folks), the gherkin syntax is very cool and easy to use, at least the way that is used in this project :D

**Comment by acasademont** [comment]
This would come handy! +1

**Comment by ciaranmcnulty** [comment]
Thanks for the contributution!

Is there anything that needs to go in the docs for this?

**Comment by Markcial** [comment]
only a slight mention about the new feature of shared namespace between specs and classes might be fine. Thanks for the merge :)

**Pull request merged**